### DISCUSSION TITLE
Data entry - drop down list possible?

### DISCUSSION THREAD

**Discussion by groston** [dbeaver/dbeaver#14106] [General]
I have just created a MySQL database using DBeaver. Good tool, no problems.

For now, until I get a simple website built, i will be entering data into the database via DBeaver itself. The values to be inserted into some of the fields in some of the tables should be taken from a drop-down list. As an example, if the column were named EyeColor, then the possible choices could be Brown, Blue, Hazel. Doing this on a webpage is easy - present an input field with a dop-down list. Is there a way to do this within DBeaver so that I do not have to type the entry each time? If so, how is this down? (and a pointer to the documentation would be great)

Thanks!

**Comment by ShadelessFox** [comment]
Hello @groston,

You can create a dictionary table containing ids and names of desired colors. Then you can use foreign key to reference it.

DBeaver provides a way to view list of every possible dictionary value for a given foreign key column:

![image](https://user-images.githubusercontent.com/35821147/135686877-2bfe164a-026b-4ff2-bc4a-c48b08020c7f.png)

**Comment by groston** [comment]
Thank you, but... I searched and found nothing about creating a dictionary value. Would you please point me to the appropriate documentation so that I can try this. Thx.

**Reply by ShadelessFox** [reply]
I believe this concept only exists within DBeaver.

Basically, dictionary table is a table consisting of a unique identifier (PK), its name, and, optionally, other data. Then you add foreign key that references this id, so it's constrained to valid values only.

An example of dictionary table:
```sql
CREATE TABLE Colors (
    ID int NOT NULL,
    Name varchar(20) NOT NULL,
    PRIMARY KEY (ID)
);
```

**Comment by groston** [comment]
ShadelessFox - Thanks, but... Yesterday, I was able to do exactly as you described and created a few rows in a table that has foreign keys defined. However, today, it is not working and nothing has really changed. I suspect that I may have turned something off. Can you point me to instructions for making this work/turn on this capability?
